hushline/settings.py

```diff
@@ -457,8 +384,53 @@ def verify_2fa_setup() -> Response | str:
         session.pop("is_setting_up_2fa", None)
         return redirect(url_for("logout"))
 
-    @bp.route("/update_pgp_key", methods=["GET", "POST"])
-    @require_2fa
+    @bp.route("/update_pgp_key_proton", methods=["POST"])
+    @authentication_required
+    def update_pgp_key_proton() -> Response | str:  # noqa: PLR0911
+        user_id = session.get("user_id")
+        if not user_id:
+            flash("⛔️ User not authenticated.")
+            return redirect(url_for("login"))
+
+        user = db.session.get(User, user_id)
+        form = PGPProtonForm()
+        if form.validate_on_submit():
+            email = form.email.data
+
+            if is_valid_email(email):
+                # Try to fetch the PGP key from ProtonMail
+                try:
+                    r = requests.get(
+                        f"https://mail-api.proton.me/pks/lookup?op=get&search={email}", timeout=5
+                    )
+                except requests.exceptions.RequestException as e:
+                    current_app.logger.error(f"Error fetching PGP key from Proton Mail: {e}")
+                    flash("⛔️ Error fetching PGP key from Proton Mail.")
+                    return redirect(url_for(".index"))
+                if r.status_code == 200:  # noqa: PLR2004
+                    pgp_key = r.text
+                    if is_valid_pgp_key(pgp_key):
+                        if user:
```

Comment:
let's move this check up to immediately after the db query? If there is no matching user, should we clear the session and redirect to the login page?